### PR TITLE
Hw2_task04

### PR DESCRIPTION
**Pull request by mirmozavr**

### CODE REVIEW

File: hw2/task04/hw2_task04.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def function(a: int, b: int) -> int:
+    return (a ** b) ** 2
+
+
+def cache(func: Callable) -> Callable:
+    f_cache = {}
+
+    def wrapper(*args: int) -> int:
```
Comment (Cjkjvfnby) [inline hw2/task04/hw2_task04.py]: Why it is int?

File: hw2/task04/test_hw2_task04.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from hw2_task04 import cache, function
+
+cache_func = cache(function)
```
Comment (Cjkjvfnby) [inline hw2/task04/test_hw2_task04.py]: Please add tests for a function `sum` and argument `[100, 200]`

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: hw2/task04/hw2_task04.py (hunk 1 above)
Comment (mirmozavr) [reply hw2/task04/hw2_task04.py]: To match data, that will be given to ''function". But cache could be used for other functions (like sum of strings) too, so I changed it to Any.

File: hw2/task04/test_hw2_task04.py (hunk 2 above)
Comment (Cjkjvfnby) [reply hw2/task04/test_hw2_task04.py]: @mirmozavr, please check the comment:

https://github.com/mirmozavr/EPAMHW/pull/9#discussion_r588637141

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
@mirmozavr, please check the comment:

https://github.com/mirmozavr/EPAMHW/pull/9#discussion_r588637141

**Comment by Cjkjvfnby** [comment]
@mirmozavr , please pay attention to the details in the comment https://github.com/mirmozavr/EPAMHW/pull/9#discussion_r588637141

File: hw2/task04/hw2_task04.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import pickle  # noqa: S403
+from typing import Any, Callable
+
+
+def function(a: int, b: int) -> int:
+    return (a ** b) ** 2
+
+
+def calc_sum(a: int, b: int) -> int:
+    return a + b
```
Comment (Cjkjvfnby) [inline hw2/task04/hw2_task04.py]: You don't need these functions here.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**